lib/qiita/events/markdown_generator.ex

```diff
@@ -0,0 +1,17 @@
+defmodule Qiita.Events.MarkdownGenerator do
+  @type items :: list()
+
+  @callback generate(items) :: String.t()
+
+  defmacro __using__(_opts) do
+    quote do
+      import Qiita.Events.Repo
+
```

Comment:
OK. If you choose to use a macro, probably you could include more common code here, such as `Qiita.Events.MarkdownGenerator`. Maybe `import Qiita.Events.TableUtils` as well if it is considered as generic. 

Other than that, all looks good to me.

Reply:
Yes.
I avoid using a macro.

But the below code happens  an error to compile.

```elixir
defmodule Qiita.Events. MarkdownGenerator do
  @type items :: list()

  @callback generate(items) :: String.t()
  
  import Qiita.Events.Repo

  def run(query, start_time, end_time, tags, title, id) do
    fetch_items(query, start_time, end_time)
    |> generate()
    |> post_markdown(tags, title, id)
  end
end
```

I desire the above code.
But I met CompileError.

```
** (CompileError) iex:10: undefined function generate/1 (expected Qiita.Events.MarkdownGenerator to define such a function or for it to be imported, but none are available)
```

I see [GenServer](https://github.com/elixir-lang/elixir/blob/v1.13.3/lib/elixir/lib/gen_server.ex#L741-L843).
So I think it uses `defmacro`.
I maybe have a misunderstanding.
Please give me a hint to solve.